### PR TITLE
Made postCheck hook required

### PR DESCRIPTION
**Pull request by ShubhSensei**
Issue #152 

**Description**

- Made _postCheck hook Require
- Made separate files for `sc_rehearsal_1`, `sc_rehearsal_2` & `sc_rehearsal_4`

### CODE REVIEW

**Comment by mds1** [comment]
Thank you! We have a few ongoing mainnet tasks at the moment, so will hold off on reviewing this until after those are all executed :)

File: script/SignFromJson.s.sol
Comment (maurelian) [inline script/SignFromJson.s.sol:39]: Looks like you accidentally deleted the `}` which is causing a compiler failure.

**Review by maurelian — CHANGES_REQUESTED** [review]
Overall the changes look sane, and now is a good time to get this merged as there are no pending playbooks in CI, however there are a few blockers:

2. a build failure I've pointed out inline
2. for each rehearsal _template_ the `justfile` will also need to be updated to replace `SignFromJson` with the new scripts added here.

We also need to figure out why CI is not running on this PR (it should be failing for the reasons above). 

In other repos I'm used to at least seeing a message indicating that a maintainer will need to take action to get the checks running. But here they are just being skipped.

**Comment by maurelian** [comment]
Got the checks running by pushing a different branch (postcheck_hook_required-2) with the same commit.

File: script/SignFromJson.s.sol
Comment (ShubhSensei) [reply script/SignFromJson.s.sol:39]: Oops, I accidentally forgot it, that's why CI is crashing, I updated it.

**Comment by maurelian** [comment]
FYI @ShubhSensei, I took your commits and am building on them over at #266

**Comment by maurelian** [comment]
Closing in favor of #266, which builds on these commits but incorporates some subtle patterns used in this repo.

**Pull request closed without merging**
